great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -58,25 +136,560 @@ def __init__(self, runtime_environment: dict):
         ):
             sys.path.append(self.plugins_directory)
 
-        # We want to have directories set up before initializing usage statistics so that we can obtain a context instance id
+        # We want to have directories set up before initializing usage statistics so
+        # that we can obtain a context instance id
         self._in_memory_instance_id = (
             None  # This variable *may* be used in case we cannot save an instance id
         )
+        # Init stores
+        self._stores = {}
+        self._init_stores(self.project_config_with_variables_substituted.stores)
+
+        # Init data_context_id
+        self._data_context_id = self._construct_data_context_id()
+
+        # Override the project_config data_context_id if an expectations_store was already set up
+        self.config.anonymous_usage_statistics.data_context_id = self._data_context_id
+        self._initialize_usage_statistics(
+            self.project_config_with_variables_substituted.anonymous_usage_statistics
+        )
+
+        # Store cached datasources but don't init them
+        self._cached_datasources = {}
+
+        # Build the datasources we know about and have access to
+        self._init_datasources()
+
+        self._evaluation_parameter_dependencies_compiled = False
+        self._evaluation_parameter_dependencies = {}
 
     @abstractmethod
     def _init_variables(self) -> DataContextVariables:
         raise NotImplementedError
 
+    @abstractmethod
+    def _save_project_config_to_disk(self) -> None:
+        """
+        Each DataContext will define how its project_config will be saved.
+            - FileDataContext : Filesystem.
+            - CloudDataContext : Cloud endpoint
+            - Ephemeral : not saved, and logging message outputted
+        """
+        raise NotImplementedError
```

Comment:
The reason I suggested to disk is to more clearly describe what this method is doing. If we have a generic method in the abstract class it can call this more descriptive method in its child implementation. I.e. `AbstractDataContext._save_project_config()` is overridden as `FileDataContext._save_project_config(): self._save_project_config_to_disk()`.

Reply:
I think I'm agreeing with @cdkini and @anthonyburdi which is this method should be called `_save_project_config` here in the abstract class. In the implementation in `FileDataContext` you can define a private method there called `_save_project_config_to_disk` and call it as @anthonyburdi describes above from there. Here, it should have a generic name that is relevant to all implementers.